mig-agent/agentcontext/env_linux.go

```diff
@@ -75,11 +75,21 @@ func findOSInfo(orig_ctx AgentContext) (ctx AgentContext, err error) {
 	}
 	logChan <- mig.Log{Desc: fmt.Sprintf("getLSBRelease() failed: %v", err)}.Debug()
 	ctx.OSIdent, err = getIssue()
-	if err == nil {
+
+	// Here we check that we read more than '\S'.
+	// See https://access.redhat.com/solutions/1138953
+	if err == nil && len(ctx.OSIdent) > 3 {
 		logChan <- mig.Log{Desc: "using /etc/issue for distribution ident"}.Debug()
 		goto haveident
 	}
 	logChan <- mig.Log{Desc: fmt.Sprintf("getIssue() failed: %v", err)}.Debug()
+
+	ctx.OSIdent, err = getOSRelease()
+	if err == nil {
+		logChan <- mig.Log{Desc: "using /etc/os-release for distribution ident"}.Debug()
+		goto haveident
+	}
+
 	logChan <- mig.Log{Desc: "warning, no valid linux os identification could be found"}.Info()
```

Comment:
Should this just return at this point?

Reply:
This function also populates the `AgentContext`'s `Init` field a couple lines down from here.